symengine/python/symengine/lib/symengine_wrapper.pyx

```diff
@@ -1698,9 +1745,374 @@ def powermod_list(a, b, m):
         s.append(c2py(<RCP[const symengine.Basic]>(v[i])))
     return s
 
+# Is this really intentional? eval_double already defined earlier:
```

Comment:
Definition at the end of the .pyx is
```
def eval_double(basic):
    cdef Basic b = sympify(basic)
    return symengine.eval_double(deref(b.thisptr))
``` 
which returns a python double.

Earlier definition was
```
def eval_double(x):
    cdef Basic X = sympify(x)
    return c2py(<RCP[const symengine.Basic]>(symengine.real_double(symengine.eval_double(deref(X.thisptr)))))
```
which returned a `SymEngine::RealDouble` and `<` operator is not specified for it. `<` operator should be defined for all `Number`s instead of `Integer`s as we have now. 

I think these two issues are better addressed in a separate PR.